AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -749,16 +755,37 @@ - (BOOL)shouldBatchFetch
   }
 }
 
-- (void)handleBatchFetchScrollingToOffset:(CGPoint)targetOffset
+- (void)_scheduleCheckForBatchFetching
 {
-  ASDisplayNodeAssert(_batchContext != nil, @"Batch context should exist");
-  
-  if (![self shouldBatchFetch]) {
+  // Push this to the next runloop to be sure the UITableView has the right content size
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [self _checkForBatchFetching];
+  });
+}
+
+- (void)_checkForBatchFetching
+{
+  // Dragging will be handled in scrollViewWillEndDragging:withVelocity:targetContentOffset:
+  if (self.isDragging || self.isTracking) {
```

Comment:
We cannot really move this check into the shared function for "should batch fetch": `_beginBatchFetchingIfNeededWithScrollView:forScrollDirection:contentOffset:` as this method will not only be called from `_checkForBatchFetching` but also from `scrollViewWillEndDragging:withVelocity:targetContentOffset:` and in there "isDragging" is actually true so it would return and not start the batch fetching.